### ISSUE TITLE
Sortable Table Header cells are not accessible via keyboard or read as clickable by screen reader

### ISSUE THREAD

**Issue by chawes13** [LaunchPadLab/lp-components#475]
The child of the `th` should likely be a button to get most of this functionality for free.

Example: https://dequeuniversity.com/library/aria/table-sortable

Raised by @kweingart08

**Comment by aalejandromr** [comment]
@chawes13 A couple of questions: 

- Should this also follow the inspiration of the arrow up/down content moved to the button instead?
- Should it also follow the inspiration of 100% width on the button?
- I do feel like the linked inspiration showcases a good practice on what the caption should have (columns + active sorted column callout). It is computed tho. Any thoughts of following that pattern? If so, it would change the current purpose of the `caption` prop. We would possibly need a migration guide callout there as well.

@marchwiany I would love to hear your thoughts here
